planning/behavior_path_planner/src/utils/start_goal_planner_common/utils.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright 2023 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "behavior_path_planner/utils/start_goal_planner_common/utils.hpp"
+
+namespace behavior_path_planner::utils::start_goal_planner_common
+{
+
+using motion_utils::calcDecelDistWithJerkAndAccConstraints;
+
+boost::optional<double> calcFeasibleDecelDistance(
+  std::shared_ptr<const PlannerData> planner_data, const double acc_lim, const double jerk_lim,
+  const double target_velocity)
+{
+  const auto v_now = planner_data->self_odometry->twist.twist.linear.x;
+  const auto a_now = planner_data->self_acceleration->accel.accel.linear.x;
+
+  if (v_now < target_velocity) {
+    return 0.0;
+  }
+
+  auto min_stop_distance = calcDecelDistWithJerkAndAccConstraints(
+    v_now, target_velocity, a_now, -acc_lim, jerk_lim, -1.0 * jerk_lim);
+
+  if (!min_stop_distance) {
+    return {};
+  }
+
+  min_stop_distance = std::max(*min_stop_distance, 0.0);
+
+  return min_stop_distance;
+}
```

Comment:
maybe this can be used in other modules. but putting here is also ok for me.

Reply:
@kosuke55 
I feel that taking `PlannerData` as an argument might be excessive if the function is defined to be used by other modules within behavior_path_planner. If there is a level of generality that could be useful across other modules and there are benefits to standardizing it, 

I would decompose the arguments to make it more common. However, I don't think there's a need to go that far. 
As a simple implementation that only needs to be usable within start/goal_planner, I've placed it in common. 

I'd like to consolidate any other simple implementations that are commonly used within start/goal_planner here as well.